SurveyApp/Services/Network/Factory/NetworkServiceFactory.swift

```diff
@@ -19,6 +19,12 @@ final class NetworkServiceFactory {
         UserSessionProvider.shared.userSession = userSession
 
         self.api = BaseAPI(userSession: userSession)
+        
+        if ProcessInfo.processInfo.environment[TestConstants.mockAPIKey] != nil {
+            let userSessionMock = UserSessionMock()
+            UserSessionProvider.shared.userSession = userSessionMock
+            self.api = SeededAPI(userSession: userSessionMock, api: api)
+        }
```

Comment:
also can do
```
func setup(userSession: UserSessionProcotol) {
self.api = BaseAPI(userSession: userSession)
}
        
```

and move if/else to AppDelegate layer.

Reply:
@blyscuit I'm a bit confused. When setting up a mock API with `SeededAPI` should it also be done in the `AppDelegate` layer or `NetworkServiceFactory` layer? 
because `SeededAPI` needs `BaseAPI` as a substituted API and only set when there is mockAPIKey:
```Swift
self.api = SeededAPI(userSession: userSession, api: api)
```
Please help suggest 🙇‍♀️

Reply:
In that case it's fine here.

Reply:
Thank you 🙏